### PR TITLE
Add API to retrieve list of project names

### PR DESCRIPTION
**Pull request by chefsalim**
Add a new API to retrieve the list of project names for a given origin.

Signed-off-by: Salim Alam <salam@chef.io>

tenor-148680636

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/builder-api/src/server/mod.rs [hunk 1]
```diff
@@ -58,14 +58,13 @@ impl HttpGateway for ApiSrv {
...
         let basic = Authenticated::new(&*config);
-        let opt = basic.clone().optional();
 
         router!(
             status: get "/status" => status,
             authenticate: get "/authenticate/:code" => github_authenticate,
 
-            job: get "/jobs/:id" => XHandler::new(job_show).before(opt.clone()),
-            job_log: get "/jobs/:id/log" => XHandler::new(job_log).before(opt.clone()),
+            job: get "/jobs/:id" => XHandler::new(job_show).before(basic.clone()),
```
Comment (reset) [inline components/builder-api/src/server/mod.rs]: tenor-256202539

**Comment by nellshamrell** [comment]
👍

**Comment by reset** [comment]
@thesentinels approve

tenor-231737622

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/282060408).

**Comment by thesentinels** [comment]
:broken_heart: Travis CI reports [this PR failed to pass the test suite](https://travis-ci.org/habitat-sh/habitat/builds/282060408).

The next step is to examine the [job](https://travis-ci.org/habitat-sh/habitat/builds/282060408) and figure out why. If it is transient, you can try re-triggering the Travis CI Job - if it passes, this PR will be automatically merged. If it is not transient, you should fix the issue and update this pull request, and issue `approve` again. If you believe it will never pass, and you are feeling :godmode:, you can issue a `force` to merge this PR anyway.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/282060408).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/282060408).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**